### PR TITLE
WIP, MAINT: remove dependence on graforvfl

### PR DESCRIPTION
**Pull request by tylerjereddy**
* `graforvfl` is still listed with a copyleft license, and its dependency stack is fairly problematic to install as noted in the code changes at gh-2. While it is only a test time dependency rather than a hard library runtime dependency, I suspect we won't be able to carry it (and its transitive deps) over to sklearn. Even independent of that, it is a bit of a pain for us to deal with long-term, so this patch attempts to remove our testing dependence on it (and its transitive deps) completely.

I could use help with:

- [x] improving the "manual" test case parameters to compensate for the loss of the "automatic" tests possible via grafo usage (this is a decent chunk of work to restore roughly the same level of testing... but we were previously overparametrized anyway..)
- [x] some `git grep` checking to make sure all the remnants of grafo.. related things are gone?

### CODE REVIEW

File: .github/constraints/deps.txt [hunk 1]
```diff
@@ -1,8 +1,4 @@
 numpy
 scipy>=1.7.1
 scikit-learn>=1.2.1
-pandas
-mealpy>=3.0.2
-permetrics>=2.0.0
```
Comment (tylerjereddy) [inline .github/constraints/deps.txt]: these are transitive deps (via grafo), so being removed here (assuming we agree to remove grafo usage..)

File: src/rvfl/tests/test_model.py [hunk 2]
```diff
@@ -365,18 +364,29 @@ def test_invalid_alpha(Classifier):
...
-    # test binary and multi-class classification against
-    # the open source graforvfl library on some synthetic
+                                      weight_scheme, alpha, exp_proba_shape,
+                                      exp_proba_median, exp_proba_min):
+    # test binary and multi-class classification against expected values
+    # from the open source graforvfl library on some synthetic
     # datasets
     X, y = make_classification(n_classes=n_classes,
-                               n_informative=8)
+                               n_informative=8, random_state=0)
```
Comment (tylerjereddy) [inline src/rvfl/tests/test_model.py]: this was a problem with the old code--we should pretty much always pin random seeds in test case generation--it wasn't being caught because both our library and grafo were getting the same varied data...

File: src/rvfl/tests/test_regression.py [hunk 3]
```diff
@@ -16,14 +14,25 @@
...
+                            weight_scheme,
+                            reg_alpha,
+                            exp_preds_shape,
+                            exp_preds_median,
+                            exp_preds_min""", [
+    # expected values are from the graforvfl library
+    (100, 10, (100,), "relu", "glorot_normal", 10, (25, 10),
+     -29.31478018, -490.5751822),
+    (100, 10, (100,), "tanh", "uniform", 100, (25, 10),
+     -34.613165002, -327.82362807),
```
Comment (tylerjereddy) [inline src/rvfl/tests/test_regression.py]: here also, I need help building up a "reasonable" number of "manual" regression test cases, obviously without fully substituting for the prior Cartesian product of cases that were being probed

**Review by tylerjereddy — COMMENTED** [review]
`git grep -E -i "grafo"` looks reasonable to me locally on this feature branch (remaining uses of the reference to that lib are fine) and the test line coverage still looks good. Of course, we're dropping our absolute number of test cases, but I think removing `grafo..` is indeed worth it and we can build up the test cases manually over time as needed without the residual licensing concerns.

Thanks for pushing in a few more cases here.

I did have a few review comments, but they're non-blocking and can be handled in a follow up.

We've both contributed here, so it isn't clear who should do the final merge. We're not actually changing any source code outside tests, so probably not a big deal. I think I'll probably self-merge in this case since you've pushed more lines than I in the end...
